examples/discovery-search-app/cypress/integration/pagination/spec.ts

```diff
@@ -0,0 +1,212 @@
+    cy.fixture('query/query.json').as('queryJSON');
+    cy.fixture('collections/collections.json').as('collectionsJSON');
+    cy.fixture('component_settings/componentSettings.json').as('componentSettingsJSON');
+    cy.fixture('query/multiPageResults.json').as('multiPageResultsJSON');
+    cy.route('GET', '**/component_settings?version=2019-01-01', '@componentSettingsJSON').as(
+      'componentSettings'
+    );
+    cy.route('POST', '**/query?version=2019-01-01', '@queryJSON').as('searchQuery');
+    cy.route('GET', '**/collections?version=2019-01-01', '@collectionsJSON').as('getCollections');
+    cy.visit('/');
+  });
+
+  describe('When the application loads', () => {
+    it('pagination component is present', () => {
+      cy.get('.bx--pagination').should('exist');
+    });
+
+    it('pagination has the correct initial settings', () => {
+      // items per page
+      cy.get('.bx--pagination__left') //TODO: change this selector to use id instead of class
+        .find('.bx--select-input')
+        .should('have.value', '5');
+      // page number
+      cy.get('.bx--pagination__right')
+        .find('.bx--select-input')
+        .should('have.value', '1');
+    });
+
+    it('items per page has the correct options available', () => {
+      itemsPerPageOptions.forEach(option => {
+        cy.get('.bx--pagination__left')
+          .find('.bx--select-input')
+          .select(option)
+          .should('have.value', option);
+      });
+    });
+  });
+
+  describe('When there are multiple pages of results', () => {
+    beforeEach(() => {
+      cy.route('POST', '**/query?version=2019-01-01', '@multiPageResultsJSON').as('multiPageQuery');
+      cy.get('.bx--search-input').type('abil{enter}');
+    });
+
+    it('lists the correct number of total pages', () => {
+      cy.get('.bx--pagination__text').should('contain', '12');
+    });
+
+    it('the previous page button is disabled', () => {
+      cy.get('.bx--pagination__button--backward').should('be.disabled');
+    });
+
+    describe('and the next page arrow is clicked', () => {
+      beforeEach(() => {
+        cy.wait('@multiPageQuery');
+        cy.get('.bx--pagination__button--forward').click();
+      });
+
+      it('correctly requests the next page', () => {
+        cy.wait('@multiPageQuery')
+          .its('requestBody.offset')
+          .should('eq', 5);
+      });
+
+      describe('and the previous page arrow is clicked', () => {
+        beforeEach(() => {
+          cy.wait('@multiPageQuery');
+          cy.get('.bx--pagination__button--backward').click();
+        });
+
+        it('correctly requests the previous page', () => {
+          cy.wait('@multiPageQuery')
+            .its('requestBody.offset')
+            .should('eq', 0);
+        });
+      });
+    });
+
+    describe('and we use the page selector', () => {
+      beforeEach(() => {
+        cy.wait('@multiPageQuery');
+      });
+
+      it('we can navigate to each page of results', () => {
```

Comment:
i think this might be a little overkill for an e2e test - i think just checking one of them should be sufficient